src/main/java/com/neatresults/mgnltweaks/json/JsonBuilder.java

```diff
@@ -431,6 +438,59 @@ public Object remove(final Object key) {
             return obj;
         }
 
+		@Override
+		public Object get(Object key) {
+			Object superResult = super.get(key);
+			if (!(superResult instanceof ContentMap))
+				return superResult;
+
+			Node node = ((ContentMap) superResult).getJCRNode();
+			if (isArrayParent(node))
+                return childrenAsContentMapList(node);
+
+            return superResult;
+		}
+
+        private boolean isArrayParent(Node candidate) {
+            for (String key : config.childrenArrayCandidates.keySet()) {
+                final String value = config.childrenArrayCandidates.get(key);
+                final Pattern valuePattern = Pattern.compile(value);
+                final Pattern keyPattern = Pattern.compile(key);
+
+                try {
+                    if (!asPropertyStream(candidate.getProperties()).filter(
+							p -> {
+								try {
+									return keyPattern.matcher(p.getName()).matches()
+											&& valuePattern.matcher(p.getString()).matches();
+								} catch (RepositoryException e) {
+									throw new RuntimeException(e);
+								}
+							}).collect(Collectors.toList()).isEmpty())
+						return true;
+                } catch (RepositoryException e) {
+                    throw new RuntimeException(e);
```

Comment:
Not big fun of a runtime exception here or elsewhere in the code either. While they look innocent here, they would eventually result in printing big ugly statement to the template generating json array or letting it not closed correctly and screwing up the JS code on the client side. I find that extremely unfriendly towards JS devs and went to big lengths to avoid it elsewhere in the code, you should try to aim for the same and recover gracefully when such thing occur. since here the only source of RE is when properties can't be obtained, I would think it ok to assume there are no useful properties and return empty array instead of throwing exception.